### PR TITLE
Use %u for unsigned, not signed

### PR DESCRIPTION
**Pull request by AtariDreams**

### CODE REVIEW

File: backend/ipp.c [hunk 1]
```diff
@@ -2759,7 +2759,7 @@ new_request(
   request = ippNewRequest(op);
   ippSetVersion(request, version / 10, version % 10);
 
-  fprintf(stderr, "DEBUG: %s IPP/%d.%d\n",
+  fprintf(stderr, "DEBUG: %s IPP/%u.%u\n",
```
Comment (michaelrsweet) [inline backend/ipp.c:2762]: These are actually defined as signed values...

File: backend/lpd.c [hunk 2]
```diff
@@ -1043,16 +1043,16 @@ lpd_queue(const char      *hostname,	/* I - Host to connect to */
       * Send the control file...
       */
 
-      if (lpd_command(fd, "\002%d cfA%03d%.15s\n", (int)strlen(control),
+      if (lpd_command(fd, "\002%lu cfA%03d%.15s\n", (unsigned long)strlen(control),
```
Comment (michaelrsweet) [inline backend/lpd.c:1046]: The control file will NEVER need the range of unsigned long. Fine making this `%u` with an unsigned cast.

File: backend/lpd.c [hunk 3]
```diff
@@ -1043,16 +1043,16 @@ lpd_queue(const char      *hostname,	/* I - Host to connect to */
...
                       (int)getpid() % 1000, localhost))
       {
 	close(fd);
 
         return (CUPS_BACKEND_FAILED);
       }
 
-      fprintf(stderr, "DEBUG: Sending control file (%u bytes)\n",
-	      (unsigned)strlen(control));
+      fprintf(stderr, "DEBUG: Sending control file (%lu bytes)\n",
```
Comment (michaelrsweet) [inline backend/lpd.c:1054]: Ditto

File: backend/usb-darwin.c [hunk 4]
```diff
@@ -772,7 +772,7 @@ print_device(const char *uri,		/* I - Device URI */
 	}
 	else if (bytes > 0)
 	{
-	  fprintf(stderr, "DEBUG: Wrote %d bytes of print data...\n", (int)bytes);
+	  fprintf(stderr, "DEBUG: Wrote %u bytes of print data...\n", (unsigned)bytes);
```
Comment (michaelrsweet) [inline backend/usb-darwin.c:775]: Since "bytes" is a signed value already, no point in changing this.

File: backend/usb-libusb.c [hunk 5]
```diff
@@ -1427,7 +1427,7 @@ open_device(usb_printer_t *printer,	/* I - Printer */
   int	number1 = -1,			/* Configuration/interface/altset */
         number2 = -1,			/* numbers */
         errcode = 0;
-  char	current;			/* Current configuration */
+  unsigned char	current;			/* Current configuration */
```
Comment (michaelrsweet) [inline backend/usb-libusb.c:1430]: Won't changing this to unsigned cause comparison warnings below?

File: backend/usb-libusb.c [hunk 6]
```diff
@@ -1521,7 +1521,7 @@ open_device(usb_printer_t *printer,	/* I - Printer */
 
   if (number1 != current)
   {
-    fprintf(stderr, "DEBUG: Switching USB device configuration: %d -> %d\n",
+    fprintf(stderr, "DEBUG: Switching USB device configuration: %u -> %d\n",
```
Comment (michaelrsweet) [inline backend/usb-libusb.c:1524]: Don't like a mix of signed/unsigned here...

File: backend/usb-unix.c [hunk 7]
```diff
@@ -195,7 +195,7 @@ void
 list_devices(void)
 {
 #ifdef __linux
-  int	i;				/* Looping var */
+  unsigned	i;				/* Looping var */
```
Comment (michaelrsweet) [inline backend/usb-unix.c:198]: Don't mix cosmetic changes in please...

**Review by michaelrsweet — CHANGES_REQUESTED** [review]
There are some problematic changes here.

**Comment by AtariDreams** [comment]
@michaelrsweet Done!

**Review by michaelrsweet — APPROVED** [review]
LGTM

**Pull request merged**